cert.go

```diff
@@ -230,6 +231,21 @@ func (c *Certificate) SetSerial(serial *big.Int) error {
 	return nil
 }
 
+// GetIssueDate retrieves the certificate issue date.
+func (c *Certificate) GetIssueDate() (time.Time, error) {
+	var ts time.Time
+	result := C.X_X509_get0_notBefore(c.x)
+	if result == nil {
+		return ts, errors.New("failed to get issue date")
+	}
+	str := C.GoString((*C.char)(unsafe.Pointer(result.data)))
```

Comment:
I think you need to cleanup the pointer, like this:
```
	ptr := unsafe.Pointer(result.data)
	defer C.free(ptr)
	str := C.GoString((*C.char)(ptr))
```

Reply:
`C.free()` is generally used to deallocate memory that was allocated using `malloc()` in C or any other C-native memory allocation functions. Since `unsafe.Pointer` is overseen by the Go runtime garbage collector, it will automatically free the memory when no longer in use. I checked the underlying `X_X509_get0_*` calls and it doesn't look like they are using `malloc()` either so we should be fine here. I will be sure to run some performance tests so that we can confirm this behavior.